internal/aws/metrics/metric_calculator.go

```diff
@@ -87,10 +87,15 @@ func (rm *MetricCalculator) Calculate(metricName string, labels map[string]strin
 
 type Key struct {
 	MetricMetadata interface{}
+	MetricType     interface{}
 	MetricLabels   attribute.Distinct
 }
 
 func NewKey(metricMetadata interface{}, labels map[string]string) Key {
```

Comment:
If there is no use case for it, then I would recommend removing it until we find an appropriate use case for it (avoid deprecate code living in the project and raise awareness later)

Reply:
This is still used in 1 or 2 other places where a metric type isnt passed in so didnt want to change those.